### PR TITLE
[librabbitmq] update to 0.14.0

### LINKED ISSUE #37733: [librabbitmq] update version to 0.14.0
### Library name

librabbitmq

### New version number

0.14.0

### Other information that may be useful (release notes, etc...)

https://github.com/alanxz/rabbitmq-c

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #37733
Update port librabbitmq to version 0.14.0
Note: no feature need to test.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/librabbitmq/fix-uwpwarning.patch [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-@@ -108,7 +108,7 @@ set(RABBITMQ_SOURCES
-     ${AMQP_SSL_SRCS}
- )
- 
--add_definitions(-DAMQP_BUILD)
-+add_definitions(-DAMQP_BUILD -D_CRT_SECURE_NO_WARNINGS)
- 
- set(RMQ_LIBRARIES ${AMQP_SSL_LIBS} ${SOCKET_LIBRARIES} ${LIBRT} ${CMAKE_THREAD_LIBS_INIT})
- 
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (JonLiu1993) [inline ports/librabbitmq/fix-uwpwarning.patch:1]: After remaking the patch, I changed the newline character from CRLF to LF.

File: ports/librabbitmq/portfile.cmake [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
   OUT_SOURCE_PATH SOURCE_PATH
   REPO alanxz/rabbitmq-c
-  REF v0.11.0 
-  SHA512 0c3dbb6e2b862e9f25e3f76df798ea272bbd81de2865950b95adf1f1e5791eb20d7c9d5a76cb7d2fda54bad5f12bdf69cbfa7e9fd1afdede6f9ec729ca2287de
+  REF "v${VERSION}" 
+  SHA512 18cb429bcfa457e359128bf458c8b9f60b1c929a8ca3a8206f40d6390d7d4c6f4c5140eb7e9ab7b401d035fc48324cbe963d058100ab65ef3faba59e7f95607e
   HEAD_REF master
   PATCHES
       fix-uwpwarning.patch
-      fix-link-header-files.patch #Remove this patch in the next version
```
Comment (JonLiu1993) [inline ports/librabbitmq/portfile.cmake:9]: The patch has been applied to the latest version.

File: ports/librabbitmq/fix-uwpwarning.patch (hunk 1 above)
Comment (BillyONeal) [reply ports/librabbitmq/fix-uwpwarning.patch:1]: Confirming: The patch in the repo was CRLF. Another pwsh casualty :/

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**